sigal/plugins/compress_assets.py

```diff
@@ -0,0 +1,146 @@
+#!/usr/bin/env python3
+
+"""Plugin to compress static files for faster HTTP transfer.
+
+Currently, 3 methods are supported:
+    - gzip. No dependency required. This is the fastest, but also largest output.
+    - zopfli. Need zopfli module from https://pypi.python.org/pypi/zopfli. gzip compatible output with optimized size.
+    - brotli. Need brotli module from https://pypi.python.org/pypi/Brotli. Brotli is the best compressor for web usage.
+
+"""
+
+from __future__ import unicode_literals
+
+import logging
+import gzip
+import shutil
+import os
+
+from sigal import signals
+from click import progressbar
+
+logger = logging.getLogger(__name__)
+
+SETTINGS = {
+        'suffixes': ['htm', 'html', 'css', 'js', 'svg'],
+        'method': 'gzip',
+        }
+
+
+class BaseCompressor:
+
+    def __init__(self, settings):
+        self.settings = settings
+        self.suffix = self.__class__.SUFFIX
```

Comment:
This is not needed. A class variable can be accessed directly via the object, so `self.suffix` works without this line. You probably just need to define it for `BaseCompressor` as well (set it to `None`).